bridges/ZLibraryBridge.php

```diff
@@ -0,0 +1,278 @@
+					'Bulgarian' => 'bulgarian',
+					'Catalan' => 'catalan',
+					'Chinese' => 'chinese',
+					'Crimean Tatar' => 'crimean',
+					'Croatian' => 'croatian',
+					'Czech' => 'czech',
+					'Danish' => 'danish',
+					'Dutch' => 'dutch',
+					'English' => 'english',
+					'Esperanto' => 'esperanto',
+					'Finnish' => 'finnish',
+					'French' => 'french',
+					'Georgian' => 'georgian',
+					'German' => 'german',
+					'Greek' => 'greek',
+					'Hebrew' => 'hebrew',
+					'Hindi' => 'hindi',
+					'Hungarian' => 'hungarian',
+					'Icelandic' => 'icelandic',
+					'Indigenous' => 'indigenous',
+					'Indonesian' => 'indonesian',
+					'Italian' => 'italian',
+					'Japanese' => 'japanese',
+					'Kazakh' => 'kazakh',
+					'Kirghiz' => 'kirghiz',
+					'Korean' => 'korean',
+					'Latin' => 'latin',
+					'Latvian' => 'latvian',
+					'Lithuanian' => 'lithuanian',
+					'Malayalam' => 'malayalam',
+					'Marathi' => 'marathi',
+					'Mongolian' => 'mongolian',
+					'Nepali' => 'nepali',
+					'Norwegian' => 'norwegian',
+					'Odia' => 'odia',
+					'Persian' => 'persian',
+					'Polish' => 'polish',
+					'Portuguese' => 'portuguese',
+					'Romanian' => 'romanian',
+					'Russian' => 'russian',
+					'Sanskrit' => 'sanskrit',
+					'Serbian' => 'serbian',
+					'Sinhala' => 'sinhala',
+					'Slovak' => 'slovak',
+					'Slovenian' => 'slovenian',
+					'Somali' => 'somali',
+					'Spanish' => 'spanish',
+					'Swahili' => 'swahili',
+					'Swedish' => 'swedish',
+					'Tajik' => 'tajik',
+					'Tamil' => 'tamil',
+					'Tatar' => 'tatar',
+					'Turkish' => 'turkish',
+					'Ukrainian' => 'ukrainian',
+					'Urdu' => 'urdu',
+					'Uzbek' => 'uzbek',
+					'Vietnamese' => 'vietnamese'
+				)
+			),
+			'extension' => array(
+				'name' => 'Extension',
+				'type' => 'list',
+				'title' => 'Only works for books',
+				'defaultValue' => '',
+				'values' => array(
+					'Any Extention' => '',
+					'pdf' => 'pdf',
+					'epub' => 'epub',
+					'djvu' => 'djvu',
+					'fb2' => 'fb2',
+					'txt' => 'txt',
+					'rar' => 'rar',
+					'mobi' => 'mobi',
+					'lit' => 'lit',
+					'doc' => 'doc',
+					'rtf' => 'rtf',
+					'azw3' => 'azw3'
+				)
+			),
+			'order' => array(
+				'name' => 'Sort order',
+				'type' => 'list',
+				'defaultValue' => 'date',
+				'values' => array(
+					'Most Popular' => '',
+					'Best Match' => 'bestmatch',
+					'Recently added' => 'date',
+					'By Title (A-Z)' => 'titleA',
+					'By Title (Z-A)' => 'title',
+					'By Year' => 'year',
+					'File Size ↓' => 'filesize',
+					'File Size ↑' => 'filesizeA'
+				)
+			)
+		),
+		'By filter' => array(),
+		'By search' => array(
+			'search' => array(
+				'name' => 'Search',
+				'required' => true
+			),
+			'exact' => array(
+				'name' => 'Exact matching',
+				'type' => 'checkbox'
+			)
+		)
+	);
+	private $name;
+	private $uri;
+
+	public function detectParameters($url) {
+		$params = array();
+
+		$regex = '/https:\/\/(b-ok\.cc|booksc\.xyz)\/s\/' .
```

Comment:
Let me know if there's a better way to keep a string of this length below 80 characters. Heredoc would still be too long.